### PR TITLE
Run ET unit tests on PT CI

### PR DESCRIPTION
**Pull request by huydhn**
This is the first PR to add all existing ET unit tests into PT CI.  The goal is to improve the coverage there to avoid breaking change from PT that could break ET.  With this, any future unit tests on ET will automatically be run on PT CI.  The duration of the job is now 40+ minutes, not too bad.

This also fixed the failed ET build in https://github.com/pytorch/pytorch/pull/123043.

Adding model coverage is a bit more evolved and requires adding new shards, so I will follow up on that in separate PRs.

[T192117506](https://www.internalfb.com/intern/tasks/?t=192117506), with the failed diffs D58295865 and D58394154

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Comment by huydhn** [comment]
About the failed tests from:

* D58295865 with `test_backends.py` and `test_backends_lifted.py`
* and D58394154 with `exir/backend/test/demos`

I see that they are not yet run on ET OSS CI https://github.com/pytorch/executorch/blob/main/pytest.ini#L40-L42, so that would still need to be done.  I assume that someone from ET with the knowledge about these tests would be able to pick that up.  They will be included in PT CI automatically once they are ready on ET.

cc @guangy10 @kit1980

File: .ci/docker/ci_commit_pins/executorch.txt [hunk 1]
```diff
@@ -1 +1 @@
-d4b3e5cc607e97afdba79dc90f8ef968142f347c
+2d35b304c1afd0a37f6cdc6b00aa767ea9b81a11
```
Comment (mergennachin) [inline .ci/docker/ci_commit_pins/executorch.txt:1]: while you're at it, can you change this to latest? 

172574a6be5910a4609e4ed1bef2b6b8475ddb3d

It contains this: https://github.com/pytorch/executorch/pull/4007

**Review by guangy10 — APPROVED** [review]

File: .ci/docker/common/install_executorch.sh [hunk 2]
```diff
@@ -37,20 +37,24 @@ install_conda_dependencies() {
...
   popd
 }
 
 setup_executorch() {
   pushd executorch
-  source .ci/scripts/utils.sh
+  # Setup swiftshader and Vulkan SDK which are required to build the Vulkan delegate
+  as_jenkins bash .ci/scripts/setup-vulkan-linux-deps.sh
 
-  install_flatc_from_source
```
Comment (mergennachin) [inline .ci/docker/common/install_executorch.sh:49]: why remove this?

Comment (huydhn) [reply .ci/docker/common/install_executorch.sh:49]: I replace it by the simpler step of calling `.ci/scripts/setup-linux.sh cmake` which will setup everything including installing flatc.  I'm trying to keep reusing the script from ET as much as I could here.  For example, if someone add another dependency in `setup-linux.sh`, the same step will be applied to PT CI right away as part of the commit hash update.

File: .ci/pytorch/test.sh [hunk 3]
```diff
@@ -1169,15 +1169,23 @@ test_executorch() {
...
-  # because it depends on PyTorch
+  export PYTHON_EXECUTABLE=python
+  export EXECUTORCH_BUILD_PYBIND=ON
+  export CMAKE_ARGS="-DEXECUTORCH_BUILD_XNNPACK=ON -DEXECUTORCH_BUILD_KERNELS_QUANTIZED=ON"
+
+  # NB: We need to rebuild ExecuTorch runner here because it depends on PyTorch
+  # from the PR
   # shellcheck disable=SC1091
   source .ci/scripts/utils.sh
+  install_executorch
```
Comment (huydhn) [inline .ci/pytorch/test.sh:1180]: I think I will replace this section by calling `.ci/scripts/setup-linux.sh cmake` like the Docker setup above

File: .ci/pytorch/test.sh [hunk 4]
```diff
@@ -1174,15 +1174,21 @@ test_executorch() {
...
+
+  # NB: We need to rebuild ExecuTorch runner here because it depends on PyTorch
+  # from the PR
   # shellcheck disable=SC1091
-  source .ci/scripts/utils.sh
-  build_executorch_runner "cmake"
+  source .ci/scripts/setup-linux.sh cmake
+
+  echo "Run ExecuTorch unit tests"
+  pytest -v -n auto
```
Comment (digantdesai) [inline .ci/pytorch/test.sh:1187]: I didn't see the failing test on D58295865, but I guess this is a start. 👍

File: .ci/docker/ci_commit_pins/executorch.txt [hunk 5]
```diff
@@ -1 +1 @@
-d4b3e5cc607e97afdba79dc90f8ef968142f347c
+172574a6be5910a4609e4ed1bef2b6b8475ddb3d
```
Comment (digantdesai) [inline .ci/docker/ci_commit_pins/executorch.txt:1]: Curious how do we update this regularly?

**Review by digantdesai — APPROVED** [review]
Looks good to me. @mergennachin ?

File: .ci/docker/ci_commit_pins/executorch.txt (hunk 5 above)
Comment (huydhn) [reply .ci/docker/ci_commit_pins/executorch.txt:1]: We have a bot to update this nightly on PT https://github.com/pytorch/pytorch/pull/123043.  The whole process is automated and the new commit is landed once ET build and test pass.  However, as you can see that the build has been failing for a while.  After this PR fixes ET build, future commit hash update will work.  On the other hand, I think we might want to setup an alert of sort here to have someone take a look when PRs like https://github.com/pytorch/pytorch/pull/123043 keep failing.

**Comment by huydhn** [comment]
As this refresh Docker, I need to remove some XFAIL python 3.12 dynamo test in `test_misc.py` to make CI green.

File: .ci/pytorch/test.sh (hunk 4 above)
Comment (huydhn) [reply .ci/pytorch/test.sh:1187]: I'm looking for someone from ET to help with that.  The good news is that one they are running on ET CI, a pinned commit update on PT will automatically bring them over to PT CI.

**Pull request closed without merging**